subsys/bluetooth/shell/bt.c

```diff
@@ -448,35 +448,41 @@ static int char2hex(const char *c, u8_t *x)
 	return 0;
 }
 
-static int str2bt_addr(const char *str, bt_addr_t *addr)
+static int hexstr2array(const char * str, u8_t * array, u8_t size)
```

Comment:
Why are you changing the coding style in addition to adding the size parameter? There should be no space between `*` and the variable name.